api/data_refinery_api/serializers.py

```diff
@@ -682,7 +682,26 @@ def validate_dataset(data):
         else:
             accessions.extend(value)
 
-    if len(accessions) > 0:
+    if len(accessions) == 0:
+        return
+
+    if data.get("quant_sf_only", False):
+        samples_without_quant_sf = Sample.public_objects.filter(
+            accession_code__in=accessions
+        ).exclude(
+            # Exclude samples that have at least one uploaded quant.sf file associated with them
+            results__computedfile__filename="quant.sf",
+            results__computedfile__s3_key__isnull=False,
+            results__computedfile__s3_bucket__isnull=False,
+        )
+        if samples_without_quant_sf.count() > 0:
+            raise serializers.ValidationError(
+                "Sample(s) '"
+                + ", ".join([s.accession_code for s in samples_without_quant_sf])
+                + "' in dataset are missing quant.sf files"
+            )
+
+    else:
```

Comment:
Hmm, I don't think this will work if the experiment specifies `ALL`. In our handling of the `ALL` keyword above we specifically check `processed_public_objects`, and return an error if we don't find any, so we wouldn't make it down here.